packages/devtools_app/lib/src/flutter/extent_delegate_list.dart

```diff
@@ -0,0 +1,464 @@
+    _offsets[0] = 0;
+    for (int i = 0; i < length; ++i) {
+      offset += computeExtent(i);
+      _offsets[i + 1] = offset;
+    }
+
+    super.recompute();
+  }
+
+  @override
+  double itemExtent(int index) {
+    if (index >= length) return 0;
+    return _offsets[index + 1] - _offsets[index];
+  }
+
+  @override
+  double layoutOffset(int index) {
+    if (index >= _offsets.length) return _offsets.last;
+    return _offsets[index];
+  }
+
+  @override
+  int minChildIndexForScrollOffset(double scrollOffset) {
+    int index = collection.lowerBound(_offsets, scrollOffset);
+    if (index == 0) return 0;
+    if (index >= _offsets.length ||
+        (_offsets[index] - scrollOffset).abs() > precisionErrorTolerance) {
+      index--;
+    }
+    assert(_offsets[index] <= scrollOffset + precisionErrorTolerance);
+    return index;
+  }
+
+  @override
+  int maxChildIndexForScrollOffset(double scrollOffset) {
+    int index = collection.lowerBound(_offsets, scrollOffset);
+    // TODO(jacobr): review.
+    if (index == 0) return 0;
+    index--;
+    assert(_offsets[index] < scrollOffset);
+    return index;
+  }
+}
+
+/// A scrollable list of widgets arranged linearly where each item has an extent
+/// specified by the [extentDelegate].
+///
+/// This class is inspired by the functionality in [ListView] where
+/// `itemExtent` is specified. The difference is the extentDelegate provided
+/// here specifies different extents for each item in the list and provides
+/// the ability to animate extents without rebuilding the list. You should use
+/// ListView instead for the simpler case where all items have the same extent.
+///
+/// Using this class is more efficient than using a ListView without specifying
+/// itemExtent as only items visible on screen need to be built and laid out.
+/// This class is more robust than ListView for cases where ListView items off
+/// screen need to be animated.
+class ExtentDelegateListView extends BoxScrollView {
+  const ExtentDelegateListView({
+    Key key,
+    Axis scrollDirection = Axis.vertical,
+    bool reverse = false,
+    ScrollController controller,
+    bool primary,
+    ScrollPhysics physics,
+    bool shrinkWrap = false,
+    EdgeInsetsGeometry padding,
+    @required this.childrenDelegate,
+    @required this.extentDelegate,
+    int semanticChildCount,
+  })  : assert(childrenDelegate != null),
+        super(
+          key: key,
+          scrollDirection: scrollDirection,
+          reverse: reverse,
+          controller: controller,
+          primary: primary,
+          physics: physics,
+          shrinkWrap: shrinkWrap,
+          padding: padding,
+          semanticChildCount: semanticChildCount,
+        );
+
+  /// A delegate that provides the children for the [ExtentDelegateListView].
+  final SliverChildDelegate childrenDelegate;
+
+  /// A delegate that provides item extents for the children of the
+  ///  [ExtentDelegateListView].
```

Comment:
super nit: remove extra space at beginning of line

Reply:
done